lib/file.js

```diff
@@ -45,17 +55,70 @@ class File {
     return !!(((mode >> 6) & 1) || (((mode << 3) >> 6) & 1) || (((mode << 6) >> 6) & 1))
   }
 
+  loadInfo () {
+    try {
+      if (this.isAppMac() && Object.keys(this.info).length === 0) {
+        //  macOS App
+        const infoPath = path.join(this.path, 'Contents', 'Info.plist')
+        this.info = plist.readFileSync(infoPath)
+        //  generate icon if it's not existed
+        this.iconPath = path.join('data', 'icons', `${this.name}-${sha1(this.path)}.png`)
+        const iconAbsPath = path.join(this.options.cwd || '', this.iconPath)
+        if (!fs.existsSync(iconAbsPath)) {
+          this.generateIcon(iconAbsPath)
+        }
+      }
+    } catch (err) {
+      console.trace(`Error when loading plist of ${this.name}: ${err}`)
+    }
+  }
+
+  generateIcon (iconPath) {
+    try {
+      //  Create directory if it's not exist.
+      const basedir = path.dirname(iconPath)
+      if (!fs.existsSync(basedir)) {
+        mkdirp.sync(basedir)
+      }
+
+      if (this.info.CFBundleIconFile) {
+        //  Generate the icon from '.icns' file
+        let icnsPath = path.join(this.path, 'Contents', 'Resources', this.info.CFBundleIconFile)
+        if (this.info.CFBundleIconFile.indexOf('.') < 0) {
+          icnsPath = `${icnsPath}.icns`
+        }
+
+        if (fs.existsSync(icnsPath)) {
+          iconutilQueue.push(icnsPath, (err, icons) => {
+            if (err) {
+              console.error(`iconutil.toIconset(${icnsPath}): ${err}`)
+            } else {
+              // TODO: remove the hardcode name
+              fs.writeFileSync(iconPath, icons['icon_128x128.png'])
```

Comment:
Sure, just update the code with `findBestIcon()` to find the icon. and with test cases.